crates/router/src/connector/payu/transformers.rs

```diff
@@ -0,0 +1,550 @@
+use error_stack::{IntoReport, ResultExt};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    core::errors,
+    pii::{self, Secret},
+    types::{self, api, storage::enums},
+    utils::OptionExt,
+};
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuPaymentsRequest {
+    customer_ip: std::net::IpAddr,
+    merchant_pos_id: String,
+    total_amount: i64,
+    currency_code: enums::Currency,
+    description: String,
+    pay_methods: PayuPaymentMethod,
+    continue_url: Option<String>,
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuPaymentMethod {
+    pay_method: PayuPaymentMethodData,
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum PayuPaymentMethodData {
+    Card(PayuCard),
+    Wallet(PayuWallet),
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum PayuCard {
+    #[serde(rename_all = "camelCase")]
+    Card {
+        number: Secret<String, pii::CardNumber>,
+        expiration_month: Secret<String>,
+        expiration_year: Secret<String>,
+        cvv: Secret<String>,
+    },
+}
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuWallet {
+    pub value: String,
+    #[serde(rename = "type")]
+    pub wallet_type: String,
+    pub authorization_code: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for PayuPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let auth_type = PayuAuthType::try_from(&item.connector_auth_type)?;
+        let payment_method = match item.request.payment_method_data.clone() {
+            api::PaymentMethod::Card(ccard) => Ok(PayuPaymentMethod {
+                pay_method: PayuPaymentMethodData::Card(PayuCard::Card {
+                    number: ccard.card_number,
+                    expiration_month: ccard.card_exp_month,
+                    expiration_year: ccard.card_exp_year,
+                    cvv: ccard.card_cvc,
+                }),
+            }),
+            api::PaymentMethod::Wallet(wallet_data) => match wallet_data.issuer_name {
+                api_models::enums::WalletIssuer::GooglePay => Ok(PayuPaymentMethod {
+                    pay_method: PayuPaymentMethodData::Wallet({
+                        PayuWallet {
+                            value: "ap".to_string(),
```

Comment:
why these are hardcoded? `ap`,`PBL`,`jp`,`PBL` these values are static and won't change?